jsettlers.mapcreator/src/jsettlers/mapcreator/main/DataTester.java

```diff
@@ -49,21 +52,48 @@
 	private final ErrorList errorList;
 	private ArrayList<Error> errors = new ArrayList<Error>();
 
+	/**
+	 * Thread for testing
+	 */
+	private Thread thread;
+
+	/**
+	 * Running flag
+	 */
+	private boolean running = true;
+
 	public DataTester(MapData data, TestResultReceiver receiver) {
 		this.data = data;
 		this.receiver = receiver;
 		errorList = new ErrorList();
 	}
 
-	public void start() {
-		new Thread(this, "data tester").start();
+	public synchronized void start() {
+		running = true;
+		this.thread = new Thread(this, "data tester");
+		thread.start();
+	}
+
+	/**
+	 * Release all resources
+	 */
+	public void dispose() {
+		running = false;
+		synchronized (this) {
+			notifyAll();
+		}
+		try {
+			thread.join(100);
```

Comment:
I have more than 1k commits, github is not counting all of them.

The DataTester was simply killed on exit because we know that we do not need it's data then. Stopping it by force may lead to corrupt DataTester data. I cannot guarantee that the rest of the code will work then.

Yes, there were thoughts about error logging but for the moment we just stuck to System.err.

We just don't have anything like a clear GUI exception handling. The GUI always attempts to catch them (if any could happen) or it simply crashes - in hope that users report the bugs they find.

Adding a good logging system would be a big improvement to the Project. #259